### ISSUE TITLE
Unnecessary CSS files in dist

### ISSUE THREAD

**Issue by artursopelnik** [withastro/astro#8567]
### Astro Info

```block
Astro                    v3.0.10
Node                     v20.5.1
System                   macOS (x64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             @astrojs/react
```

### If this issue only occurs in one browser, which browser is a problem?

Not browser-related

### Describe the Bug

When I update Astro from version 3.0.8 to 3.0.10, Vite bundles CSS files in my _astro directory in dist. These files are still bundled in the latest version, but the CSS files were never assigned. It's a bug.

## Screenshot
https://ibb.co/bvrvD3D

### What's the expected result?

No unnecessary CSS files that are not included in the dist.

### Link to Minimal Reproducible Example

#

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by lilnasy** [comment]
Thanks for opening this issue.

Due to a Vite behavior (#8329), CSS files had to be included (#8351).

Since the extra files are small and don't break anything, this would be low priority.

If someone would like to work on this issue, bluwy provided helpful pointers in #8351.

**Comment by pratik-codes** [comment]
Hey @lilnasy I would like to give it a shot

**Comment by lilnasy** [comment]
Thanks @pratik-codes
Go for it!

**Comment by pratik-codes** [comment]
`When I update Astro from version 3.0.8 to 3.0.10, and run ´npm run dev´, Astro creates unnecessary CSS files in dist folder. These files are still bundled in the latest version, but the CSS files were never assigned. It's definitely a bug`

To check this bug I wanted to reproduce this so I created a new astro project and did the same thing. But it wasn't creating an unnecessary css file in the dist folder. 

To be sure about it I removed all the CSS from the project and when built it didn't even create a CSS file. 
<img width="473" alt="Screenshot 2023-09-25 at 11 40 31 PM" src="https://github.com/withastro/astro/assets/64960569/bc3a7b99-98bc-416b-9191-6bfbaf62557d">


Attaching a screenshot above. 

@lilnasy I think there is no issue or do you think I am doing anything wrong here?

**Comment by lilnasy** [comment]
That is probably a typo - `npm run dev` does **not** affect the `dist` folder - we are conserved with `npm run build`.

**Comment by pratik-codes** [comment]
@lilnasy correct I understood it was a typo. I tried npm run build only and it didn't generate anything when there was no CSS in Dist. I will also share a video for better understanding of this.

**Comment by cprass** [comment]
This happens for me too when building the static page. The CSS in the extra file comes from a local `<style>` definition in an `.astro` component. The same CSS is both inlined in the HTML file and provided as a separate CSS.

Wouldn't it be generally better to use the separate CSS file instead of the inline CSS? When using a strong CSP header, inlining CSS is not allowed. Loading it from the same origin would be fine.

**Comment by artursopelnik** [comment]
Now it works as expected with Astro 3.2. Topic can be closed.

**Comment by artursopelnik** [comment]
@cprass You're right. I still had v3.0.8. In version 3.2.2 the bug is still there.

(PR)[https://github.com/withastro/astro/tree/css-away] from @bluwy works for me.
